File: CRUD/PHP/Symfony3 V2/Symfony3/src/Sistema/Bundle/Controller/EstadisticasController.php
<?php

// Written By Ismael Heredia in the year 2017

namespace Sistema\Bundle\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\Controller;

class EstadisticasController extends Controller
{
    public function indexAction()
    {
        if($this->get("sistema.seguridad")->validar_session()) {
            
            $nombre_usuario = $this->get('sistema.seguridad')->recibirUsuarioEnSession();

            $em = $this->getDoctrine()->getManager();

            $conexion = $this->get("sistema.conexion");
            $productos = $conexion->listarProductos("");
            $cantidad_productos = count($productos);

            $array_textos_grafico1 = array();
            $array_series_grafico1 = array();

            $array_textos_grafico2 = array();
            $array_series_grafico2 = array();

            $query = $em->createQuery("SELECT prod.nombre,prod.precio FROM SistemaBundle:Productos prod JOIN SistemaBundle:Proveedores prov WHERE prod.idProveedor=prov.id");

            $resultados = $query->getResult();   
            
            foreach($resultados as $resultado) {
                $nombreProducto = $resultado["nombreProducto"];
                $precio = $resultado["precio"];
                $serie  = array(
                    'name' => $nombreProducto,
                    'y' => (int) $precio
                );
                array_push($array_textos_grafico1,$nombreProducto);
                array_push($array_series_grafico1,$serie);
            }

            $textos_grafico1 = json_encode($array_textos_grafico1);
            $series_grafico1 =  json_encode($array_series_grafico1);

            $query = $em->createQuery("SELECT prov.nombre,COUNT(prod.idProveedor) as cantidad FROM SistemaBundle:Productos prod JOIN SistemaBundle:Proveedores prov WHERE prod.idProveedor=prov.id GROUP BY prov.nombre");

            $resultados = $query->getResult();   
            
            foreach($resultados as $resultado) {
                $nombreEmpresa = $resultado["nombre"];
                $cantidad = $resultado["cantidad"];
                $serie  = array(
                    'name' => $nombreEmpresa,
                    'y' => (int) $cantidad
                );
                array_push($array_textos_grafico2,$nombreEmpresa);
                array_push($array_series_grafico2,$serie);
            }

            $textos_grafico2 = json_encode($array_textos_grafico2);
            $series_grafico2 =  json_encode($array_series_grafico2);

            return $this->render("SistemaBundle:Estadisticas:reporte.html.twig",array("nombre_usuario"=>$nombre_usuario,"productos"=>$productos,"cantidad_productos"=>$cantidad_productos,"textos_grafico1"=>$textos_grafico1,"series_grafico1"=>$series_grafico1,"textos_grafico2"=>$textos_grafico2,"series_grafico2"=>$series_grafico2));

        } else {
            return $this->redirect($this->generateUrl("sistema_login"));
        }
 
    }
}

File: CRUD/Java usando NetBeans/Aplicacion Web usando MVC Spring V2/SistemaWebSpring/src/main/java/sistema/controllers/EstadisticasController.java
// Written by Ismael Heredia in the year 2017
//SELECT prod.id,prod.nombre.prod.precio,prov.nombre AS proveedor,prod.fecha_registro FROM productos prod,proveedores prov WHERE prod.id_proveedor=prov.id
//SELECT prod.nombre,SUM(precio) AS precio FROM productos prod,proveedores prov where prod.id_proveedor=prov.id GROUP BY prod.nombre
//SELECT prov.nombre,count(prod.id_proveedor) AS cantidad FROM productos prod,proveedores prov WHERE prod.id_proveedor=prov.id GROUP BY prov.nombre
package sistema.controllers;

import javax.servlet.http.HttpServletRequest;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;
import sistema.functions.Funciones;

@Controller
public class EstadisticasController {

    Funciones funcion = new Funciones();

    @RequestMapping("/administracion/estadisticas")
    public ModelAndView FormEstadisticasListado(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            ModelAndView model = new ModelAndView("estadisticas/reporte");
            return model;
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

}
